ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
+ */
+static inline uintptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
 {
-  return (*((intptr_t *) &name) << 1) | 0x1;
+    uintptr_t tmp, sentinel = 0x1;
+    /* local jobid must fit in 15 bits */
+    assert(! (ORTE_LOCAL_JOBID(name.jobid) & 0x8000));
+    tmp = (uintptr_t)ORTE_LOCAL_JOBID(name.jobid);
+    sentinel |= ((tmp << 1) & 0xfffe);
```

Comment:
I've been trying to understand this logic, and I'm not sure I believe that the "1" you want to set for the sentinel is winding up in the correct place. Isn't it supposed to be in the MSB position?